dbcmetrics/dbcm_logic/dbcstorage.py

```diff
@@ -0,0 +1,177 @@
+from typing import Dict
+from typing import List
+from prometheus_client import Gauge, Info
+import boto3
+from botocore.config import Config
+
+
+class StorageMetricsThreading(object):
+    '''
+    Module to get metrics about an object storage
+    '''
+    def __init__(self):
+        self.storage_interval = int(os.getenv("STORAGE_INTERVAL"))
+        self.storage_exclude_subfolders = os.getenv("STORAGE_EXCLUDE_SUBFOLDERS").lower()
+        self.storage_provider_url = os.getenv("STORAGE_PROVIDER_URL")
+        self.storage_provider_region = os.getenv("STORAGE_PROVIDER_REGION")
+        self.bucket_name = os.getenv("BUCKET_NAME")
+        self.access_key = os.getenv("ACCESS_KEY")
+        self.access_secret = os.getenv("ACCESS_SECRET")
+
+        logging.info("Connecting with Access Key {} to S3 region {} by using the storage provider url: {}".format(self.access_key, self.storage_provider_region, self.storage_provider_url))
+        logging.info("Exclude subfolders when generating metrics: {}".format(self.storage_exclude_subfolders))
+
+        self.s3_client = boto3.client(
+            's3',
+            region_name=self.storage_provider_region,
+            aws_access_key_id=self.access_key,
+            aws_secret_access_key=self.access_secret,
+            endpoint_url=self.storage_provider_url,
+        )
+
+        buckets = self.s3_client.list_buckets()['Buckets']
+        logging.debug("For Access Key {} available S3 Buckets: {}".format(self.access_key, buckets))
+
+        self.bucket_availability_gauge = Gauge('storage_bucket_availability','Indicates if the target bucket is available',[
+            'name',
+            'storage_provider_url',
+            'access_key',
+            ])
+
+        self.size_bucket_gauge = Gauge('storage_size_bucket','The total size in bytes of all files in a bucket',[
+            'name',
+            'storage_provider_url',
+            'access_key',
+            ])
+
+        self.file_count_bucket_gauge = Gauge('storage_file_count_bucket','The total number of files in a bucket',[
+            'name',
+            'storage_provider_url',
+            'access_key',
+            ])
+
+        self.size_folder_gauge = Gauge('storage_size_folder','The total size in bytes of all files in a folder',[
+            'name',
+            'bucket',
+            'storage_provider_url',
+            'access_key',
+            ])
+
+        self.file_count_folder_gauge = Gauge('storage_file_count_folder','The total number of files in a folder',[
+            'name',
+            'bucket',            
+            'storage_provider_url',
+            'access_key',
+            ])
+
+        self.thread = threading.Thread(target=self.run)
+        self.thread.daemon = True
+        self.thread.start()
+
+    def run(self):
+        '''
+        Function that calls metric functions and waits in a loop until the stop event is send from the main thread
+        '''
+        def do_something():
```

Comment:
Isn't there a better name?

Reply:
I think so, it's just to keep consistency with the `dbcversion.py`. But I could either break consistency and choose a better name in the `dbcstorage.py` or keep consistency but modify a part outside the scope of the ticket by changing the name in the `dbcstorage.py` as well as `dbcversion.py`.

Reply:
I solved the issue (following the zen of python "simple is better than complex") by removing the `do_something()` method all together because it didn't serve a purpose, as it neither improved readability nor functionality.